extra/Lamdera/Injection.hs

```diff
@@ -171,582 +171,80 @@ source mode mains =
-          eum: function() { upgradeMode = true },
-          die: die,
-          fns: {}
-        } : {};
-      }
-
     var isLamderaRuntime = typeof isLamdera !== 'undefined';
 
-    function _Platform_initialize(flagDecoder, args, init, update, subscriptions, stepperBuilder)
-      {
-        var result = A2(_Json_run, flagDecoder, _Json_wrap(args ? args['flags'] : undefined));
-
-        // @TODO need to figure out how to get this to automatically escape by mode?
-        //$$elm$$core$$Result$$isOk(result) || _Debug_crash(2 /**/, _Json_errorToString(result.a) /**/);
-        $$elm$$core$$Result$$isOk(result) || _Debug_crash(2 /**_UNUSED/, _Json_errorToString(result.a) /**/);
+    function _Lamdera_inject(app, callUpdate, model) {
+      app.die = app.stop;
 
+      app.fns =
+        { decodeWirePayloadHeader: $$author$$project$$LamderaHelpers$$decodeWirePayloadHeader
+        , decodeWireAnalytics: $$author$$project$$LamderaHelpers$$decodeWireAnalytics
+        , getUserModel : function() { return model.userModel }
+        };
 
-        var managers = {};
-        var initPair = init(result.a);
-        var model = (args && args['model']) || initPair.a;
-
-        var stepper = stepperBuilder(sendToApp, model);
-        var ports = _Platform_setupEffects(managers, sendToApp);
-
-        var pos = 0;
+      var pos = 0;
 
-        function mtime() { // microseconds
-          if (!isLamderaRuntime) { return 0; }
-          const hrTime = process.hrtime();
-          return Math.floor(hrTime[0] * 1000000 + hrTime[1] / 1000);
-        }
-
-        var buriedTimestamp = null;
+      function mtime() { // microseconds
+        if (!isLamderaRuntime) { return 0; }
+        const hrTime = process.hrtime();
+        return Math.floor(hrTime[0] * 1000000 + hrTime[1] / 1000);
+      }
 
-        function sendToApp(msg, viewMetadata)
-        {
-          if (buriedTimestamp !== null) {
-            const elapsed = Date.now() - buriedTimestamp;
-            // This tries to turn `HomePageMsg (WeatherWidgetMsg (WeatherReportReceived WeatherReport))`
-            // into `"HomePageMsg WeatherWidgetMsg WeatherReportReceived"`.
-            // The idea is that if the timeout for forwarding messages isn't enough, we want to know what
-            // message somebody used that took even longer, but without reporting the entire msg.
-            // Note that in `--optimize` mode, the above string would become something like `"1 3 6"`,
-            // but it's better than nothing.
-            let msgName = '(unknown message)';
-            if (msg.$) {
-              msgName = msg.$;
-              let current = msg;
-              while (current.a && current.a.$ && !current.b) {
-                current = current.a;
-                msgName = msgName + ' ' + current.$;
-              }
-            }
-            bugsnag.notify(new Error('Got message ' + elapsed + ' ms after app was buried: ' + msgName));
-            return;
-          }
```

Comment:
@supermario Has this ever happened in Bugsnag? Was it worth it? I have a feeling we don’t need this anymore, so I haven’t ported it. What do you think?

Reply:
Just checked and nope, looks like we've never gotten this bugsnag yet... but this feels like one of those things that's not that big a deal until it happens, and then not having it will be very confusing. Perhaps just being overly paranoid. Happy to leave it for now 👍